x/merkledb/helpers_test.go

```diff
@@ -0,0 +1,219 @@
+package merkledb
+
+import (
+	"bytes"
+
+	"github.com/ava-labs/avalanchego/ids"
+	pb "github.com/ava-labs/avalanchego/proto/pb/sync"
+	"github.com/ava-labs/avalanchego/utils/maybe"
+)
+
+func protoProofsEqual(proof1, proof2 *pb.Proof) bool {
```

Comment:
I feel like we should just use `require.Equal`. I'd rather the UT take a bit longer than have to maintain this additional code, and I think it's more readable to just use `require.Equal` in the tests.

Reply:
It makes the tests take 4x as long though and we are hitting the timeout.

Reply:
If you feel that strongly about it, I'll make them fuzz tests instead and still use require.equal